File: lib/script/impl/src/testIntegration/resources/hamal/scope/0002.hs
-- local
local x = 10
do
    assert(x == 10)
    do
        assert(x == 10)
        local x = 20
        assert(x == 20)
    end
    assert(x == 10)
end
assert(x == 10)

--global
g = 10
do
    assert(g == 10)
    do
        assert(g == 10)
        g = 20
        assert(g == 20)
    end
    assert(g == 20)
end
assert(g == 20)

File: faas/mono/src/integrationTest/resources/suite/func/0002.lua
local sys = require('sys')

local err, funcs = sys.func.list()
assert(err == nil)
assert(#funcs == 0)

local err, func_one_req = sys.func.create({ name = 'func-1' })
sys.await(func_one_req)

assert(err == nil)
assert(func_one_req ~= nil)
--
local _, funcs = sys.func.list()
assert(#funcs == 1)

assert(func_one_req.id == funcs[1].id)
assert(funcs[1].name == 'func-1')

err, func_two_req = sys.func.create({ name = 'func-2' })
sys.await(func_two_req)

_, funcs = sys.func.list()
assert(#funcs == 2)
